packages/db/src/project/index.ts

```diff
@@ -5,80 +5,147 @@ import type { Provider } from "web3/providers";
-    const { db, project: input } = options;
+export async function initialize(options: InitializeOptions): Promise<Project> {
+  const { run, forProvider } =
+    "db" in options ? Process.Run.forDb(options.db) : options;
 
-    const { run, forProvider } = forDb(db);
+  const { project: input } = options;
 
-    const project = await run(generateInitializeLoad, input);
+  const project = await run(Initialize.process, { input });
 
-    return new Project({ run, forProvider, project });
-  }
+  return new Project({ run, forProvider, project });
+}
 
-  get id() {
+/**
+ * Abstraction for connecting @truffle/db to a Truffle project
+ *
+ * This class affords an interface between Truffle and @truffle/db,
+ * specifically for the purposes of ingesting @truffle/workflow-compile results
+ * and contract instance information from @truffle/contract-schema artifact
+ * files.
+ *
+ * Unless you are building tools that work with Truffle's various packages
+ * directly, you probably don't need to use this class.
+ *
+ * @example To instantiate this abstraction:
+ * ```typescript
+ * import { connect, Project } from "@truffle/db";
+ *
+ * const db = connect({
+ *   // ...
+ * });
+ *
+ * const project = await Project.initialize({
+ *   db,
+ *   project: {
+ *     directory: "/path/to/project/dir"
+ *   }
+ * });
+ * ```
+ */
+export class Project {
+  public get id(): string {
     return this.project.id;
   }
 
   /**
    * Accept a compilation result and process it to save all relevant resources
-   * (Source, Bytecode, Compilation, Contract)
+   * ([[DataModel.Source | Source]], [[DataModel.Bytecode | Bytecode]],
+   * [[DataModel.Compilation | Compilation]],
+   * [[DataModel.Contract | Contract]])
+   *
+   * This returns the same WorkflowCompileResult but with additional
+   * references to each of the added resources.
+   *
+   * @category Truffle-specific
    */
   async loadCompile(options: {
     result: WorkflowCompileResult;
   }): Promise<{
-    compilations: Compilation[];
-    contracts: Contract[];
+    compilations: LoadCompile.Compilation[];
+    contracts: LoadCompile.Contract[];
   }> {
     const { result } = options;
 
-    return await this.run(generateCompileLoad, result);
+    return await this.run(LoadCompile.process, result);
   }
 
   /**
    * Update name pointers for this project. Currently affords name-keeping for
-   * Network and Contract resources (e.g., naming ContractInstance resources
-   * is not supported directly)
+   * [[DataModel.Network | Network]] and [[DataModel.Contract | Contract]]
+   * resources (e.g., naming [[DataModel.ContractInstance | ContractInstance]]
+   * resources is not supported directly)
+   *
+   * This saves [[DataModel.NameRecord | NameRecord]] and
+   * [[DataModel.ProjectName | ProjectName]] resources to @truffle/db.
+   *
+   * Returns a list of NameRecord resources for completeness, although these
+   * may be regarded as an internal concern. ProjectName resources are not
```

Comment:
```suggestion
   * may be regarded as an internal concern. `ProjectName` resources are not
```

Reply:
I haven't been putting those in backticks... roughly my style has been to add a hyperlink for the first usage in a section, then just style it as normal text subsequently. Think we should use a different style?